eureka-common/src/main/java/edu/emory/cci/aiw/cvrg/eureka/common/props/AppPropertiesRegistration.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package edu.emory.cci.aiw.cvrg.eureka.common.props;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Miao Ai
+ */
+public class AppPropertiesRegistration {
+	private boolean localAccountRegistrationEnabled = false;
+	private boolean registrationEnabled = false;
+	private String googleOAuthKey;
+	private String googleOAuthSecret;
+	private String gitHubOAuthKey;
+	private String gitHubOAuthSecret;
+	private String twitterOAuthKey;
+	private String twitterOAuthSecret;
+	private String globusOAuthKey;
+	private String globusOAuthSecret;        
+	/**
+	 * Constructor.
+	 *
+	 */    
+	AppPropertiesRegistration(){
+                AppProperties appProperties = AppProperties.getInstance();
+                localAccountRegistrationEnabled = (Boolean.parseBoolean(appProperties.getValue("eureka.webapp.localregistrationenabled")));
+                googleOAuthKey = appProperties.getValue("eureka.webapp.googleoauthkey");
+                googleOAuthSecret = appProperties.getValue("eureka.webapp.googleoauthsecret");
+                gitHubOAuthKey = appProperties.getValue("eureka.webapp.githuboauthkey");
+                gitHubOAuthSecret = appProperties.getValue("eureka.webapp.githuboauthsecret");
+                twitterOAuthKey = appProperties.getValue("eureka.webapp.twitteroauthkey");
+                twitterOAuthSecret = appProperties.getValue("eureka.webapp.twitteroauthsecret");
+                globusOAuthKey = appProperties.getValue("eureka.webapp.globusoauthkey");
+                globusOAuthSecret = appProperties.getValue("eureka.webapp.globusoauthsecret");                
+                
+        } 
+        
+	public boolean isRegistrationEnabled() {
+		return isLocalAccountRegistrationEnabled() || isOAuthRegistrationEnabled();
+	}   
+	/**
+	 * Gets the localAccountRegistrationEnabled .
+	 *
+	 * @return localAccountRegistrationEnabled.
+	 */        
+	public boolean isLocalAccountRegistrationEnabled() {
+		return localAccountRegistrationEnabled;
+	}        
+	/**
+	 * Sets the localAccountRegistrationEnabled status.
+	 *
+	 * @param inLocalAccountRegistrationEnabled.
+	 */  
+	public void setLocalAccountRegistrationEnabled(boolean inLocalAccountRegistrationEnabled) {
+                this.localAccountRegistrationEnabled = inLocalAccountRegistrationEnabled;
+	}   
+        
+ 
+	public List<String> getOAuthRegistrationProviders(){
```

Comment:
Make the strings returned by this method an enum. I'm pretty sure that you can make this method return a list of the enum values and Jersey will automatically convert them to strings.